src/clr-angular/utils/focus/key-focus/key-focus.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+  AfterContentChecked,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Inject,
+  Input,
+  Output,
+  OnInit,
+  PLATFORM_ID,
+} from '@angular/core';
+
+import { isPlatformBrowser } from '@angular/common';
+
+import { Observable } from 'rxjs';
+
+@Directive({
+  selector: '[clrKeyFocus]',
+})
+export class ClrKeyFocus implements OnInit, AfterContentChecked {
+  constructor(private elementRef: ElementRef, @Inject(PLATFORM_ID) private platformId: Object) {}
+
+  @Input('clrKeyFocus') direction: 'horizontal' | 'vertical' | 'both';
+  @Input('clrFocusOnLoad') focusOnLoad: boolean = false;
+
+  private _focusChange: EventEmitter<void> = new EventEmitter();
+
+  @Output('clrFocusChanged')
+  get focusChange(): Observable<void> {
+    return this._focusChange.asObservable();
+  }
+
+  current: number = 0;
+
+  ngOnInit() {
+    if (!this.direction) {
+      this.direction = 'vertical';
+    }
+    if (this.focusOnLoad && isPlatformBrowser(this.platformId)) {
+      this.getCurrent().focus();
+      this._focusChange.next();
+    }
+  }
+
+  ngAfterContentChecked() {
+    this.getChildren().forEach(button => {
+      this.setTabIndex(button, -1);
+    });
+    this.setTabIndex(this.getCurrent(), 0);
+  }
+
+  @HostListener('keydown', ['$event'])
+  handleKeyboardEvent(event: any) {
+    if (this.prevClicked(event) && this.current - 1 >= 0) {
+      this.keyAction(() => {
+        this.current--;
+      });
+    } else if (this.nextClicked(event) && this.current + 1 < this.getChildren().length) {
+      this.keyAction(() => {
+        this.current++;
+      });
+    } else if (event.code === 'Home') {
+      this.keyAction(() => {
+        this.current = 0;
+      });
+    } else if (event.code === 'End') {
+      this.keyAction(() => {
+        this.current = this.getChildren().length - 1;
+      });
+    }
+  }
+
+  @HostListener('click', ['$event'])
+  handleClickEvent(event: any) {
+    const position = this.getPositionOf(event.srcElement as HTMLElement);
+    if (position > -1) {
+      this.current = position;
+    }
+  }
+
+  getPositionOf(element: HTMLElement) {
+    const children = Array.from(this.getChildren());
```

Comment:
The NodeList returned by getChildren does not have indexOf.
Probably could move inside getChildren, as its return type is wrong now anyway.